tf_model/train_classifier.py

```diff
@@ -0,0 +1,265 @@
+        alpha_char = tf.strings.regex_replace(remove_contraction, '[^a-zA-Z.]', ' ')
+        remove_outer_whitespace = tf.strings.strip(alpha_char)
+        remove_extra_whitespace = tf.strings.regex_replace(remove_outer_whitespace, '\s{2,}', ' ')
+        
+        return tf.strings.lower(remove_extra_whitespace)
+    
+    
+    def create_vectorizer(self, dataset: np.ndarray) -> TextVectorization:
+        
+        vectorizer = TextVectorization(
+            max_tokens=self.vocab_size,
+            standardize=self.preprocess_tweet,
+            output_mode='int',
+            output_sequence_length=self.max_length,
+            name='vectorizer')
+        
+        vectorizer.adapt(dataset)
+
+        return vectorizer
+    
+    
+    def create_emb_matrix(self, vectorizer):
+        
+        vocab = vectorizer.get_vocabulary()
+        word_index = dict(zip(vocab, range(len(vocab))))
+        
+        emb_matrix = np.zeros((self.vocab_size, self.embed_dim))
+        for word, index in word_index.items():
+            vector = glove_vectors.get(word)
+            if vector is not None:
+                emb_matrix[index] = vector
+                
+        return emb_matrix
+        
+        
+    def call(self, inputs, training=False):
+        
+        data, labels = self.get_data()
+        train_data, train_labels, val_data, val_labels, _, _ = self.data_splits(data, labels)
+        vectorizer = self.create_vectorizer(data)
+        emb_matrix = self.create_emb_matrix(vectorizer)
+        
+        tweet_input = Input(shape=(1,), dtype=tf.string, name='tweet')
+        
+        x = vectorizer(tweet_input)
+        
+        x = Embedding(input_dim=self.vocab_size,
+                      output_dim=self.embed_dim,
+                      input_length=self.max_length,
+                      weights=[emb_matrix],
+                      trainable=False,
+                      name='GloVe_200')(x)
+        
+        x_1 = self.bigram(x)
+        x_1 = self.pool(x_1)
+
+        x_2 = self.trigram(x)
+        x_2 = self.pool(x_2)
+
+        x_3 = self.fourgram(x)
+        x_3 = self.pool(x_3)
+
+        x_4 = self.fivegram(x)
+        x_4 = self.pool(x_4)
+        
+        merged = self.concat([x_1, x_2, x_3, x_4])
+        
+        merged = self.dense(merged)
+        
+        if training:
+            merged = self.dropout(merged)
+
+        predictions = self.preds(merged)
+        
+        model = Model(tweet_input, predictions)
+
+        return model
+
+
+    def compile_model(self, model):
+
+        if self.num_classes == 2:
+            model.compile(loss='binary_crossentropy', optimizer='adam', metrics=['accuracy'])
+    
+        else:
+            model.compile(
+            loss='sparse_categorical_crossentropy', optimizer='adam', metrics=['sparse_categorical_accuracy'])
+    
+        return model
+
+
+    def custom_callbacks(self):
+
+        es = EarlyStopping(
+            monitor='val_sparse_categorical_accuracy',
```

Comment:
if you're parametrizing based on number of classes, the custom callbacks `monitor` attribute will need to be parametrized too